### PR TITLE
[Azure Search] Add Get Documents API (a.k.a. Lookup) to data plane API spec

### PR DESCRIPTION
**Pull request by bitsforthought**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/brjohnstmsft/azure-rest-api-specs/generate-lookup).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by brjohnstmsft** [comment]
Big thanks to @mhko who did the majority of the work on this.

**Comment by brjohnstmsft** [comment]
@jhendrixMSFT Some context on this change: This is not a new API, it's just that until now we've never generated code for it. This is the first of many steps we're taking to make this particular Swagger spec accurately reflect our data plane API.

**Comment by jhendrixMSFT** [comment]
@brjohnstmsft do you do any local C# codegen to verify this looks ok?

File: specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/examples/SearchIndexAutocompleteDocumentsPost.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
           "highlightPostTag": "</em>",
           "highlightPreTag": "<em>",
           "minimumCoverage": 80,
-          "searchFields": ["title", "description"],
+          "searchFields": "title,description",
```
[the remark this answers is not in the document]
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/examples/SearchIndexAutocompleteDocumentsPost.json]: @nschonni That's the spec for the GET version of Autocomplete. The example I updated was for POST. AutoRest uses array parameters to represent query string parameters that are comma-separated lists. So although it's a comma-separated list everywhere on the wire, in the spec it's a string for POST and an array of strings for GET.

**Comment by brjohnstmsft** [comment]
@jhendrixMSFT Yes, I'm working on a .NET SDK PR that I will send out once this is merged. I wrote the translation rules in tandem with changing the .NET SDK.

File: specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/examples/SearchIndexAutocompleteDocumentsPost.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (nschonni) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/examples/SearchIndexAutocompleteDocumentsPost.json]: Gotcha, I guess it's the alias'd property here https://github.com/Azure/azure-rest-api-specs/blob/576cce05e63736d742d3ab389b9c41a2cf98852c/specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/searchindex.json#L518-L522

[the remark this answers is not in the document]
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/examples/SearchIndexAutocompleteDocumentsPost.json]: Almost, it's this one: https://github.com/Azure/azure-rest-api-specs/blob/576cce05e63736d742d3ab389b9c41a2cf98852c/specification/search/data-plane/Microsoft.Azure.Search.Data/preview/2017-11-11-preview/searchindex.json#L546

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**